### PR TITLE
Improve XSD date handling (#1181)

### PR DESCRIPTION
**Pull request by giacomociti**
distinguish date and dateTime according to #1181

### CODE REVIEW

**Comment by ovatsus** [comment]
Yes, command line build and CI is currently broken, though building and running tests in VS still works

File: src/Xml/XmlRuntime.fs [hunk 1]
```diff
@@ -208,7 +208,7 @@ type XmlRuntime =
             | _ ->
                 match v with
                 | :? string         as v -> v
-                | :? DateTime       as v -> v.ToString("O", cultureInfo)
+                | :? DateTime       as v -> v.ToString("yyyy-MM-dd")
```
Comment (ovatsus) [inline src/Xml/XmlRuntime.fs]: Why this change?

**Comment by ovatsus** [comment]
Can you update one of the xsd test files used in SignatureTestCases.config and regenerate the expected files so we have coverage for this? (you can regenerate by evaluating SignatureTests.fs in FSI and running generateAllExpected()

File: src/Xml/XmlRuntime.fs (hunk 1 above)
Comment (giacomociti) [reply src/Xml/XmlRuntime.fs]: If we make `System.DateTime` correspond to `xs:date` (and `System.DateTimeOffset` correspond to `xs:dateTime`) when formatting a `xs:date` we have to avoid the time component in order to produce valid xml according to the XML Schema specification. On the other hand, when formatting a `xs:dateTime` it's ok to use the `"O"` format that is compliant with the W3C specs for `xs:dateTime`

Comment (giacomociti) [reply src/Xml/XmlRuntime.fs]: The problem is, even if we emit zeros for the time component, the resulting xml would be invalid.

Maybe we could use the "yyyy-MM-dd" format when the time component is zero and use the "O" format otherwise.

**Comment by ovatsus** [comment]
Would you mind also updating the contributing.md adding that note about generateAllExpected?

File: src/Xml/XmlRuntime.fs (hunk 1 above)
Comment (ovatsus) [reply src/Xml/XmlRuntime.fs]: That would work

**Comment by ovatsus** [comment]
Looks good, can you update release notes and add a test that asserts the toString output doesn't have the time component?

**Comment by ovatsus** [comment]
Thanks
